core/src/test/java/org/apache/calcite/test/verifier/SmtLibTest.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.calcite.test.verifier;
+
+
+import com.microsoft.z3.BoolExpr;
+import com.microsoft.z3.Context;
+import com.microsoft.z3.Expr;
+import com.microsoft.z3.Solver;
+import com.microsoft.z3.Status;
+
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+/**
+ * Simple Test for testing dynamic library for z3.
+ **/
+
+public class SmtLibTest {
+  @Test void testZ3Lib1() {
+    try {
+      /** need to specify the local native library **/
+      //System.load("/usr/local/lib/libz3java.dylib");
+      Context z3Context = new Context();
+      Expr x = z3Context.mkIntConst("x");
+      Expr one = z3Context.mkInt(1);
+      BoolExpr eq = z3Context.mkEq(x, one);
+      Solver solver = z3Context.mkSolver();
+      solver.add(eq);
+      assertEquals(solver.check(), Status.SATISFIABLE);
+    } catch (Error e) {
+      System.err.println("cannot load native library libz3java ");
```

Comment:
Please move "z3 availability" check to `@BeforeAll` method, so all the tests are marked as skipped in case the library is not available.

The test could `throw Exception`, so you don't need to catch exceptions.